packages/framework/fluid-framework/src/index.ts

```diff
@@ -19,31 +19,56 @@ export { ConnectionState } from "@fluidframework/container-loader";
 export type {
 	ContainerAttachProps,
 	ContainerSchema,
-	DataObjectClass,
 	IConnection,
 	IFluidContainer,
 	IFluidContainerEvents,
 	IMember,
 	InitialObjects,
 	IServiceAudience,
 	IServiceAudienceEvents,
-	LoadableObjectClass,
-	LoadableObjectClassRecord,
 	MemberChangedListener,
 	Myself,
 } from "@fluidframework/fluid-static";
+export type { SharedObjectKind, ISharedObjectEvents } from "@fluidframework/shared-object-base";
 
 // Let the tree package manage its own API surface, we will simply reflect it here.
 // Note: this only surfaces the `@public` API items from the tree package. If the `@beta` and `@alpha` items are
 // desired, they can be added by re-exporting from one of the package's aliased export paths instead (e.g. `tree
 // alpha` to surface everything `@alpha` and higher).
-// eslint-disable-next-line no-restricted-syntax
+// eslint-disable-next-line no-restricted-syntax, import/export
 export * from "@fluidframework/tree";
 
+import type { SharedObjectKind } from "@fluidframework/shared-object-base";
+import { SharedTree as OriginalSharedTree } from "@fluidframework/tree/internal";
+import { type ITree } from "@fluidframework/tree";
+
+/**
+ * A hierarchical data structure for collaboratively editing strongly typed JSON-like trees
+ * of objects, arrays, and other data types.
+ * @public
+ */
+// eslint-disable-next-line import/export
+export const SharedTree: SharedObjectKind<ITree> = OriginalSharedTree;
+
 // ===============================================================
 // Legacy exports
 
-export { ContainerErrorTypes } from "@fluidframework/container-definitions/internal";
+export {
+	ContainerErrorTypes,
+	type IDeltaManager,
+	type IDeltaManagerEvents,
+	type IDeltaSender,
+	type IDeltaQueue,
+	type ReadOnlyInfo,
+	type IConnectionDetails,
+	type IDeltaQueueEvents,
+} from "@fluidframework/container-definitions/internal";
```

Comment:
due to our API extractor configuration, these types had to be exported. I'm not clear on the exact cause, but with how package boundaries are handled its not surprising.

Reply:
This would imply that we added something _new_ to the API here that required pulling all those things in. Can we not just remove whatever that was? I really don't think we want to expose the delta manager types if we can avoid it.

Reply:
> fluid-framework: Error: /home/craig/Work/FluidFramework.git/proxy/packages/dds/shared-object-base/src/sharedObject.ts:145:2 - (ae-forgotten-export) The symbol "IDeltaManager" needs to be exported by the entry point index.d.ts

So thats from SharedObjectCore.

> fluid-framework: Error: /home/craig/Work/FluidFramework.git/proxy/packages/dds/shared-object-base/src/sharedObject.ts:626:1 - (ae-forgotten-export) The symbol "SharedObjectCore" needs to be exported by the entry point index.d.ts

So thats from SharedObject.

> fluid-framework: Error: /home/craig/Work/FluidFramework.git/proxy/packages/dds/sequence/src/sequence.ts:116:1 - (ae-forgotten-export) The symbol "SharedObject" needs to be exported by the entry point index.d.ts

Which we need because of SharedSegmentSequence. Which was already exported.

No change was made to the relevant exports which depend on these things.

What did change is we explicitly export something @fluidframework/shared-object-base, which causes to have to export all direct dependencies of existing exports that are from that package, and all the things they reference in packages we explicitly export from recursively.

While this could be fixed by not exporting anything from @fluidframework/shared-object-base, I don't think thats a good solution as we want to export SharedObjectKind, and moving it to another package to doge this issue would be silly.

I tried to clean this up by requiring all transitive referenced types to be exported in https://github.com/microsoft/FluidFramework/pull/20554 to avoid this mess and make thing much more clear, but I was unable to get support for that change.

Thus I don't see what I should do other than what I did here, which I agree is bad and confusing. I'm open to suggestions.

Reply:
`ae-forgotten-exports` is **not** currently enabled in this package, right? Or was that enabled at some point?

If any of these types were already transitively referenced, then I'm concerned that they weren't already an issue. If they are newly transitively referenced, then I think we need to find a solution that doesn't leak them. Or we can re-disable the forgotten exports rule until we fixed things here. I would definitely prefer the former, but the latter is probably fine if we follow up quickly.

@jason-ha FYI

Reply:
Based on the docs in common/build/build-common/api-extractor-lint.json, I think we suppress ae-forgotten-export giving it log level none, but thats only about cross package stuff? The config isn't super clear to me.